api/apps/api/test/projects/project-summaries/output-project-summaries.e2e-spec.ts

```diff
@@ -398,10 +399,13 @@ const getFixtures = async () => {
       await projectRepo.insert({ id: projectId, name, organizationId: orgId });
       return projectId;
     },
-    GivenScenarioWithAtLeastOneRunExistsOnAPI: async (
+    GivenScenarioRanAtLeastOnceExistsOnAPI: async (
       name: string,
       projectId: string,
+      numberOfRuns?: number,
     ) => {
+      // Even though numberOfRuns shouldn't be null, it can happen very sporadically because of MRXN-29
+      // so ranAtLeastOnce must be used as criteria to select the scenarios for output summary
```

Comment:
😬 I think my explanation in MRXN23-474 was indeed confusing: `numberOfRuns` _has nothing to do_ with what we are checking here: `numberOfRuns` is a Marxan setting, that we use to instruct the Marxan solver about how many iterations to use when running its solver flow.

It does not tell us "how many times a user has run a scenario successfully" - we do record this information only through API events, but we don't really expose it in any way through the API or use it internally. It is only a Marxan setting, but does not describe any state of things.

The fix in the present PR is not needed because `numberOfRuns` number may be zero or undefined: it is needed because it should never be used to infer whether a scenario has ever run successfully; the only thing we should ever check is that if a scenario's `ranAtLeastOnce` property is true, then it should be included in the summary.

I recommend to remove this comment altogether, and in terms of tests, I suggest to remove the handling of `numberOfRuns` from `GivenScenarioRanAtLeastOnceExistsOnAPI()`, and adjust fixtures below to only deal with `ranAtLeastOnce`.

tl;dr we should not be concerned at all with `numberOfRuns` in this PR: setting it explicitly to zero should not even be something we ever do in tests or in reality, unless we want to test that Marxan will still default to its default number of runs if it is supplied a number that doesn't make sense (but this would be testing Marxan, not our app, so I'd forget about this altogether 😅)


```suggestion
```

Reply:
Ahhh I see what you mean, I wasn't aware of that property being a setting per se. Yeah, I agree, better to not leave that info on there to avoid further confusion.